[Jules]: fix typescript errors

### Task Requirements

You are an Expert React and TypeScript Developer specializing in Home Assistant custom frontend panels. Your task is to resolve a series of strict TypeScript compilation errors that are causing our CI/CD pipeline to fail during the Vite build process.

<detected_errors>
Error: src/App.tsx(1,38): error TS6133: 'useRef' is declared but its value is never read.
Error: src/App.tsx(4,24): error TS2307: Cannot find module './components/DeviceView' or its corresponding type declarations.
Error: src/App.tsx(7,28): error TS2307: Cannot find module './utils/api' or its corresponding type declarations.
Error: src/App.tsx(35,9): error TS6133: 'finalConfigEntryId' is declared but its value is never read.
Error: src/App.tsx(228,93): error TS2322: Property 'class' does not exist... Did you mean 'className'?
Error: src/components/EventLog.tsx(2,28): error TS2307: Cannot find module '../utils/api'
Error: src/components/HaSwitch.tsx(40,7): error TS2322: Property 'disabled' does not exist
Error: src/components/SSIDView.tsx(24,10): error TS6133: 'isToggling' is declared but its value is never read.
Error: src/components/SSIDView.tsx(70,28): error TS2448: Block-scoped variable 'isToggling' used before its declaration.
Error: src/components/SSIDView.tsx(81,17): error TS2322: Property 'disabled' does not exist
Error: src/components/TimedAccess.tsx(129,11): error TS6133: 'net' is declared but its value is never read.
Error: src/entry.tsx(29,26): error TS2322: Property 'config_entry_id' does not exist on type 'IntrinsicAttributes & AppProps'.
Error: src/meraki-panel.ts(135,11): error TS6133: '_moreInfo' is declared but its value is never read.
</detected_errors>

<instructions>
1. Check the requirements to prevent regressions and update requirements based on implemented features.
2. Think step-by-step. Analyze the provided source files to identify where each TypeScript error originates.
3. Outline your strategy inside <thinking> tags. Map out the fixes for unused variables, JSX attribute typos, missing interface properties, and scoping issues.
4. Remove all unused variable declarations and unused imports to satisfy `noUnusedLocals: true`.
5. Replace all instances of the HTML `class=` attribute with the React `className=` attribute in your JSX code.
6. Extend the interfaces (e.g., `AppProps`, `HaSwitchProps`) to explicitly define missing properties like `config_entry_id: string` and `disabled?: boolean`.
7. Correct block-scoping errors by declaring variables like `isToggling` before they are accessed in the code.
8. Verify all import paths (e.g., `./components/DeviceView` and `./utils/api`) match the exact case-sensitive filenames on disk. Add `.js` or `.ts` extensions if required by your tsconfig module resolution.
9. Output the fully refactored code for each modified file inside its own <output_code file="path/to/file.tsx"> tag.
</instructions>

<examples>
Example 1: Fixing unused variables and JSX attributes
Input:
import React, { useState, useRef } from 'react';

const MyComponent = () => {
  const unusedVar = "hello";
  return <div class="my-style">Test</div>;
};

Output:
import React, { useState } from 'react';

const MyComponent = () => {
  return <div className="my-style">Test</div>;
};

Example 2: Fixing missing interface properties
Input:
interface ButtonProps {
  label: string;
}
const CustomButton = (props: ButtonProps) => {
  return <button disabled={props.disabled}>{props.label}</button>; // Error: disabled does not exist
};

Output:
interface ButtonProps {
  label: string;
  disabled?: boolean;
}
const CustomButton = (props: ButtonProps) => {
  return <button disabled={props.disabled}>{props.label}</button>;
};
</examples>

Please ask me to provide the contents of `src/App.tsx`, `src/entry.tsx`, `src/components/SSIDView.tsx`, `src/components/HaSwitch.tsx`, and `src/meraki-panel.ts` inside <input_code> tags to begin.

### Specific Constraints

Ignore all errors not related to the meraki_ha integration. Follow Home Assistant Sentence Case standards.

--- a/custom_components/meraki_ha/www/dev_server.log
+++ b/custom_components/meraki_ha/www/dev_server.log
@@ -1,12 +1,12 @@
 
-> meraki-ha-www@2.2.0-beta.29 dev
-> vite --host 0.0.0.0
+> meraki-ha-www@2.3.0-beta.120 dev
+> vite --host
 
 
 (!) build.outDir must not be the same directory of root or a parent directory of root as this could cause Vite to overwriting source files with build outputs.
 
 
-  VITE v5.4.21  ready in 240 ms
+  VITE v5.4.21  ready in 253 ms
 
   ➜  Local:   http://localhost:5173/
   ➜  Network: http://192.168.0.2:5173/

--- a/custom_components/meraki_ha/www/src/App.tsx
+++ b/custom_components/meraki_ha/www/src/App.tsx
@@ -1,4 +1,4 @@
-import React, { useState, useEffect, useRef } from 'react';
+import React, { useState, useEffect } from 'react';
 // Frontend version: 2.3.0-beta.120
 import NetworkView from './components/NetworkView';
 import DeviceView from './components/DeviceView';
@@ -11,13 +11,14 @@ interface MerakiData {
   [key: string]: any;
 }
 
-// Update props to accept hass and panel
+// Update props to accept hass, panel, and config_entry_id
 interface AppProps {
   hass: any;
-  panel: any;
+  panel?: any;
+  config_entry_id?: string;
 }
 
-const App: React.FC<AppProps> = ({ hass, panel }) => {
+const App: React.FC<AppProps> = ({ hass, panel, config_entry_id }) => {
   const [data, setData] = useState<MerakiData | null>(null);
   const [loading, setLoading] = useState<boolean>(true);
   const [error, setError] = useState<string | null>(null);
@@ -29,11 +30,7 @@ const App: React.FC<AppProps> = ({ hass, panel }) => {
   const [showSettings, setShowSettings] = useState(false);
   const [showTimedAccess, setShowTimedAccess] = useState(false);
 
-  const configEntryId = panel?.config?.config_entry_id;
-
-  // If we are in standalone mode, config_entry_id might come from window
-  const finalConfigEntryId =
-    configEntryId || (window as any).CONFIG_ENTRY_ID;
+  const configEntryId = config_entry_id || panel?.config?.config_entry_id;
 
   useEffect(() => {
     if (window.location.hostname === 'localhost') {
@@ -157,6 +154,7 @@ const App: React.FC<AppProps> = ({ hass, panel }) => {
     }
 
     fetchData();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [configEntryId]); // Rerun if configEntryId changes
 
   const fetchData = async () => {
@@ -225,7 +223,7 @@ const App: React.FC<AppProps> = ({ hass, panel }) => {
   if (configNotFound) {
     return (
       <div className="flex flex-col justify-center items-center h-screen bg-light-background dark:bg-dark-background text-light-text dark:text-dark-text p-4 text-center">
-        <ha-icon icon="mdi:alert-circle-outline" style={{'--mdc-icon-size': '64px'} as any} class="mb-4 text-red-500"></ha-icon>
+        <ha-icon icon="mdi:alert-circle-outline" style={{'--mdc-icon-size': '64px'} as any} className="mb-4 text-red-500"></ha-icon>
         <h2 className="text-xl font-bold mb-2">Integration Not Configured</h2>
         <p className="mb-6 max-w-md">
           The Meraki integration has not been configured yet, or the configuration entry could not be found.
@@ -244,7 +242,7 @@ const App: React.FC<AppProps> = ({ hass, panel }) => {
   if (error && !data) {
     return (
       <div className="flex flex-col justify-center items-center h-screen bg-light-background dark:bg-dark-background text-light-text dark:text-dark-text p-4 text-center">
-        <ha-icon icon="mdi:error-outline" style={{'--mdc-icon-size': '64px'} as any} class="mb-4 text-red-500"></ha-icon>
+        <ha-icon icon="mdi:error-outline" style={{'--mdc-icon-size': '64px'} as any} className="mb-4 text-red-500"></ha-icon>
         <h2 className="text-xl font-bold mb-2">Error</h2>
         <p className="mb-6">{error}</p>
         <button
@@ -303,7 +301,6 @@ const App: React.FC<AppProps> = ({ hass, panel }) => {
           configEntryId={configEntryId}
         />
       ) : (
-        /* @ts-ignore */
         <DeviceView
           activeView={activeView}
           setActiveView={setActiveView}

--- a/custom_components/meraki_ha/www/src/components/EventLog.tsx
+++ b/custom_components/meraki_ha/www/src/components/EventLog.tsx
@@ -19,6 +19,10 @@ interface EventLogProps {
   productTypes?: string[];
 }
 
+interface EventLogResponse {
+  events: MerakiEvent[];
+}
+
 const EventLog: React.FC<EventLogProps> = ({
   hass,
   networkId,
@@ -62,7 +66,7 @@ const EventLog: React.FC<EventLogProps> = ({
           throw new Error('Hass connection not available');
         }
 
-        const resultData = await safeCallWS(hass, {
+        const resultData = await safeCallWS<EventLogResponse>(hass, {
           type: 'meraki_ha/get_network_events',
           config_entry_id: configEntryId,
           network_id: networkId,

--- a/custom_components/meraki_ha/www/src/components/SSIDView.tsx
+++ b/custom_components/meraki_ha/www/src/components/SSIDView.tsx
@@ -21,7 +21,7 @@ const SSIDView: React.FC<SSIDViewProps> = ({
   configEntryId,
 }) => {
   const [localSSIDs, setLocalSSIDs] = useState(ssids);
-  const [isToggling, setIsToggling] = useState<Record<number, boolean>>({});
+  const [togglingStates, setTogglingStates] = useState<Record<number, boolean>>({});
 
   useEffect(() => {
     // Update local state if ssids prop changes
@@ -31,7 +31,7 @@ const SSIDView: React.FC<SSIDViewProps> = ({
   const handleToggle = async (ssidNumber: number, enabled: boolean) => {
     if (!hass || !configEntryId) return;
 
-    setIsToggling((prev) => ({ ...prev, [ssidNumber]: true }));
+    setTogglingStates((prev) => ({ ...prev, [ssidNumber]: true }));
 
     const updatedSSIDs = localSSIDs.map((ssid) =>
       ssid.number === ssidNumber ? { ...ssid, enabled: enabled } : ssid
@@ -51,7 +51,7 @@ const SSIDView: React.FC<SSIDViewProps> = ({
       // Revert UI state on error
       setLocalSSIDs(ssids); // Revert to original state from props
     } finally {
-      setIsToggling((prev) => ({ ...prev, [ssidNumber]: false }));
+      setTogglingStates((prev) => ({ ...prev, [ssidNumber]: false }));
     }
   };
 
@@ -67,7 +67,7 @@ const SSIDView: React.FC<SSIDViewProps> = ({
     <div className="mt-4 grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4">
       {displayedSSIDs.map((ssid) => {
         const isEnabled = ssid.enabled || (ssid.entity_id && hass?.states?.[ssid.entity_id]?.state === 'on');
-        const isToggling = isToggling[ssid.number];
+        const isCurrentlyToggling = togglingStates[ssid.number];
 
         return (
           <div
@@ -78,7 +78,7 @@ const SSIDView: React.FC<SSIDViewProps> = ({
               <span className="font-medium text-lg">{ssid.name}</span>
               <ha-switch
                 checked={isEnabled}
-                disabled={isToggling}
+                disabled={isCurrentlyToggling}
                 onClick={(e: any) => {
                   e.stopPropagation(); // Prevent triggering parent onClick if any
                   handleToggle(ssid.number, !isEnabled);

--- a/custom_components/meraki_ha/www/src/components/TimedAccess.tsx
+++ b/custom_components/meraki_ha/www/src/components/TimedAccess.tsx
@@ -45,6 +45,7 @@ const TimedAccess: React.FC<TimedAccessProps> = ({
   // Initial load
   useEffect(() => {
     fetchKeys();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
   // Fetch policies when network changes
@@ -52,12 +53,13 @@ const TimedAccess: React.FC<TimedAccessProps> = ({
     if (selectedNetwork) {
       fetchPolicies(selectedNetwork);
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [selectedNetwork]);
 
   const fetchKeys = async () => {
     setLoading(true);
     try {
-      const result = await safeCallWS(hass, {
+      const result = await safeCallWS<TimedAccessKey[]>(hass, {
         type: 'meraki_ha/timed_access/get_keys',
         config_entry_id: configEntryId,
       });
@@ -71,7 +73,7 @@ const TimedAccess: React.FC<TimedAccessProps> = ({
 
   const fetchPolicies = async (networkId: string) => {
     try {
-      const result = await safeCallWS(hass, {
+      const result = await safeCallWS<GroupPolicy[]>(hass, {
         type: 'meraki_ha/timed_access/get_policies',
         config_entry_id: configEntryId,
         network_id: networkId,
@@ -126,7 +128,6 @@ const TimedAccess: React.FC<TimedAccessProps> = ({
 
   // Helper to find SSIDs for selected network
   const getSsidsForNetwork = (networkId: string) => {
-    const net = data?.networks?.find((n: any) => n.id === networkId);
     // Ideally filter for IPSK without Radius.
     // We assume data.ssids or data.networks contains SSIDs details.
     // The main App.tsx passes `data` which has `ssids`. But those are flat list of ALL SSIDs?

--- a/custom_components/meraki_ha/www/src/global.d.ts
+++ b/custom_components/meraki_ha/www/src/global.d.ts
@@ -1,3 +1,8 @@
+declare module '*.css?inline' {
+  const content: string;
+  export default content;
+}
+
 declare namespace JSX {
   interface IntrinsicElements {
     'ha-card': React.DetailedHTMLProps<
@@ -11,6 +16,7 @@ declare namespace JSX {
     'ha-switch': React.DetailedHTMLProps<
       React.HTMLAttributes<HTMLElement> & {
         checked?: boolean;
+        disabled?: boolean;
         onchange?: (e: any) => void;
       },
       HTMLElement

--- a/custom_components/meraki_ha/www/src/meraki-panel.ts
+++ b/custom_components/meraki_ha/www/src/meraki-panel.ts
@@ -132,15 +132,6 @@ export class MerakiPanel extends LitElement {
     }
   }
 
-  private _moreInfo(entityId: string) {
-    const event = new Event('hass-more-info', {
-      bubbles: true,
-      composed: true,
-    });
-    (event as any).detail = { entityId };
-    this.dispatchEvent(event);
-  }
-
   render() {
     if (this._loading) {
       return html`<p>Loading...</p>`;
@@ -154,7 +145,7 @@ export class MerakiPanel extends LitElement {
       return html`<p>No data found.</p>`;
     }
 
-    const { networks, devices, version, enabled_networks } = this._data;
+    const { networks, version, enabled_networks } = this._data;
 
     return html`
       <ha-card header="Meraki Dashboard">

